### PR TITLE
Use methods to set read marks

### PR DESCRIPTION
**Pull request by xzhflying**
Use methods to set read marks for the creator and updater in course announcements

### CODE REVIEW

File: app/models/course/announcement.rb [hunk 1]
```diff
@@ -15,4 +15,14 @@ def set_defaults
     self.start_at ||= Time.zone.now
     self.end_at ||= 7.days.from_now
   end
+
+  # Mark announcement as read for the creator
```
Comment (lowjoel) [inline app/models/course/announcement.rb]: Should be private.

Comment (lowjoel) [inline app/models/course/announcement.rb]: All of these callback methods should be private.

Comment (xzhflying) [inline app/models/course/announcement.rb]: Ok, I thought there should be a `private` above the `set_defaults` :smile:

Comment (lowjoel) [inline app/models/course/announcement.rb]: There should be, but there isn't, please add in :)

Comment (xzhflying) [inline app/models/course/announcement.rb]: Done :smile_cat: and sorry that I'm careful enough...

**Comment by lowjoel** [comment]
Rebase please.

**Comment by lowjoel** [comment]
Oh, you should write specs for this.

**Comment by xzhflying** [comment]
This model has specs at the beginning, but it seems they are too complicated. I've tried to modify them.

File: spec/models/course/announcement_spec.rb [hunk 2]
```diff
@@ -6,70 +6,60 @@
...
-
-    context 'when title is not present' do
-      subject { build(:course_announcement, title: '') }
+    describe 'create an announcement' do
+      context 'when announcement is created' do
+        subject { Course::Announcement.new }
 
-      it { is_expected.not_to be_valid }
-    end
+        it { is_expected.not_to be_sticky }
```
Comment (lowjoel) [inline spec/models/course/announcement_spec.rb]: you've changed the meaning of the spec.

Comment (xzhflying) [inline spec/models/course/announcement_spec.rb]: hmm...？I just modified the order of some specs and delete some useless codes...

Comment (xzhflying) [inline spec/models/course/announcement_spec.rb]: Maybe it's better to view this file in split model on github?

Comment (lowjoel) [inline spec/models/course/announcement_spec.rb]: no, you've definitely changed the meaning of the specs. ask @allenwq.

File: spec/models/course/announcement_spec.rb [hunk 3]
```diff
@@ -6,70 +6,60 @@
...
+    let!(:user) { create(:user) }
+    let(:course) { create(:course) }
 
-      it { is_expected.not_to be_sticky }
-    end
-
-    context 'when title is not present' do
-      subject { build(:course_announcement, title: '') }
+    describe 'create an announcement' do
+      context 'when announcement is created' do
```
Comment (allenwq) [inline spec/models/course/announcement_spec.rb]: Your specs are scoped incorrectly as I think. Why the first 2 context are all related to `'create an announcement'` ?   

And why you create a course and user for this ?

Comment (allenwq) [inline spec/models/course/announcement_spec.rb]: `'create an announcement'` and `''when announcement is created' are kind of duplications

Comment (xzhflying) [inline spec/models/course/announcement_spec.rb]: First I think these two contexts should be tested when create an announcement. Secondly, `course` and `user` are mainly used in unread status tests in both create and edit announcements.

File: spec/models/course/announcement_spec.rb (hunk 2 above)
Comment (allenwq) [inline spec/models/course/announcement_spec.rb]: @lowjoel I'm not sure if I get you, or do you mean the specs for `unread status` ?

**Pull request merged**